File: backend/app/schemas/risk.py
from pydantic import BaseModel, ConfigDict, Field


class RiskOverview(BaseModel):
    highRiskCount: int
    mediumRiskCount: int
    lowRiskCount: int
    avgRiskScore: float
    avgOperationalRisk: float
    avgEsgRisk: float
    avgOverallRisk: float

    model_config = ConfigDict(from_attributes=True)


class RiskHistogramBin(BaseModel):
    label: str
    start: float
    end: float
    count: int

    model_config = ConfigDict(from_attributes=True)


class RiskSegmentationItem(BaseModel):
    riskLevel: str
    supplierCount: int

    model_config = ConfigDict(from_attributes=True)


class RiskSupplierItem(BaseModel):
    supplierId: int
    supplierName: str
    country: str | None = None
    category: str | None = None
    tier: str | None = None
    avgDelay: float
    avgDefect: float
    avgCostVariance: float
    operationalRiskScore: float
    esgRiskScore: float
    overallRiskScore: float
    riskScore: float
    operationalRiskLevel: str
    esgRiskLevel: str
    riskLevel: str

    model_config = ConfigDict(from_attributes=True)


class DueDiligenceRequest(BaseModel):
    supplierId: int = Field(gt=0)


class DueDiligenceResponse(BaseModel):
    caseId: str | None = None
    supplier: str
    supplierId: int | None = None
    country: str | None = None
    tier: str | None = None
    status: str | None = None
    opRisk: str
    opRiskScore: float
    esgRisk: str
    esgRiskScore: float
    overall: str
    overallRiskScore: float
    decision: str | None = None
    decisionRationale: list[str] = []
    investigationChecklist: list[dict] = []
    riskDrivers: list[dict] = []
    evidenceGaps: list[str] = []
    recommendedActions: list[str] = []
    connectedSignals: dict = {}
    issues: list[str]
    aiSummary: str
    ai_source: str | None = None
    ai_provider: str | None = None
    ai_model: str | None = None
    ai_trace_id: str | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/risk.ts
import { apiRequest } from "./client";

export interface RiskOverview {
  highRiskCount: number;
  mediumRiskCount: number;
  lowRiskCount: number;
  avgRiskScore: number;
  avgOperationalRisk: number;
  avgEsgRisk: number;
  avgOverallRisk: number;
}

export interface RiskHistogramBin {
  label: string;
  start: number;
  end: number;
  count: number;
}

export interface RiskSegmentationItem {
  riskLevel: "High" | "Medium" | "Low";
  supplierCount: number;
}

export interface RiskSupplierItem {
  supplierId: number;
  supplierName: string;
  country: string | null;
  category: string | null;
  tier: string | null;
  avgDelay: number;
  avgDefect: number;
  avgCostVariance: number;
  operationalRiskScore: number;
  esgRiskScore: number;
  overallRiskScore: number;
  riskScore: number;
  operationalRiskLevel: "High" | "Medium" | "Low";
  esgRiskLevel: "High" | "Medium" | "Low";
  riskLevel: "High" | "Medium" | "Low";
}

export interface DueDiligenceResponse {
  caseId: string | null;
  supplier: string;
  supplierId: number | null;
  country: string | null;
  tier: string | null;
  status: string | null;
  opRisk: string;
  opRiskScore: number;
  esgRisk: string;
  esgRiskScore: number;
  overall: string;
  overallRiskScore: number;
  decision: string | null;
  decisionRationale: string[];
  investigationChecklist: Array<{ label: string; status: string; detail: string }>;
  riskDrivers: Array<{ label: string; value: number; status: string }>;
  evidenceGaps: string[];
  recommendedActions: string[];
  connectedSignals: Record<string, unknown>;
  issues: string[];
  aiSummary: string;
}

export async function getRiskOverview(): Promise<RiskOverview> {
  return apiRequest<RiskOverview>("/risk/overview");
}

export async function getRiskDistribution(): Promise<RiskHistogramBin[]> {
  return apiRequest<RiskHistogramBin[]>("/risk/distribution");
}

export async function getRiskSegmentation(): Promise<RiskSegmentationItem[]> {
  return apiRequest<RiskSegmentationItem[]>("/risk/segmentation");
}

export async function getTopRiskSuppliers(): Promise<RiskSupplierItem[]> {
  return apiRequest<RiskSupplierItem[]>("/risk/top-suppliers");
}

export async function runDueDiligence(supplierId: number): Promise<DueDiligenceResponse> {
  return apiRequest<DueDiligenceResponse>("/risk/due-diligence", {
    method: "POST",
    json: { supplierId },
  });
}
